docs:#823 Clarify encryption docs

micheleRP opened the pull request:
resolves #823 
resolves #1045 

This replaces the encryption prereq (which linked to Juan's personal gist and had errors) with updated steps from Denis.

preview: https://deploy-preview-1148--redpanda-documentation.netlify.app/docs/manage/security/encryption/

micheleRP commented:
Hi @dcoded and @patrickangeles! Please review the updated Prerequisites section for TLS/mTLS, and please add anyone else you think should review it.

Feediver1 commented on docs/manage/security/encryption.mdx:
```diff
@@ -4,22 +4,148 @@ title: Configure Encryption
+
+#### Create node certificate configuration
+
+A CNF file may need to be made for each node to modify the `alt_names` section. For production usage, update the `distinguished_name` section.
```
```suggestion
You might need to create a CNF file for each node to modify the `alt_names` section. For production usage, update the `distinguished_name` section.
```

Feediver1 commented on docs/manage/security/encryption.mdx:
````diff
@@ -4,22 +4,148 @@ title: Configure Encryption
+openssl req -new -config node.cnf -key node.key -out node.csr
+```
+
+If done by a corporate CA, these certificate signing requests must be signed with the following extensions:
````
```suggestion
If generated by a corporate CA, these certificate signing requests must be signed with the following extensions:
```

Feediver1 reviewed (approved):
LGTM

micheleRP commented:
@dcoded, @patrickangeles: Could you approve this?

micheleRP commented:
status = Denis ran into issues testing this. Waiting for Patrick to have time to test.

patrickangeles commented:
Some comments:
- Maybe better titled as **Configuring Kafka TLS Encryption** (so as not to be confused with on-disk encryption or something else)
- This covers client-to-broker communications via the Kafka API... I haven't seen how you setup TLS for broker-to-broker RPC communications. Is that covered in another section?
- Some sections could use further context. For example: why is this step optional? **Optional: Create a local CA for self-signed certificates**
- Because of `# comments` for the `openssl` commands, copy/paste doesn't work. Suggestion: break out the comments so that they're in normal document text, not part of the copy/paste block (those should be just for the commands).

General comment: overall it reads somewhere between a HOWTO and a reference doc... it might be clearer to break these out into separate pages? Or structure this page somehow so that we go through what needs to be configured, but also show some of the example steps but be clear that they're examples.

Related is that the examples seems to assume a host based installation... you can't just edit `redpanda.yaml` for a container based install. So the examples won't apply to any user trying this on a docker based setup (which is a likely scenario if you're following a HOWTO and trying this on your local machine).

micheleRP commented on docs/manage/security/encryption.mdx:
```diff
@@ -1,27 +1,167 @@
+| --------            | ------    | 
+| `-new`                | New request.    | 
+| `-config node.cnf`    | Configuration file to use when generating certificates (created above).    | 
+| `-key ca.key`         | Private key of the node (created above).    | 
```
```suggestion
| `-key node.key`         | Private key of the node (created above).    | 
```
@patrickangeles the code above uses `-key node.key`. Should I change this to match?

micheleRP commented on docs/manage/security/encryption.mdx:
```diff
@@ -1,27 +1,167 @@
+
+| Output             | Description   |
+| --------            | ------    | 
+| `-out node.crt`     | The public key certificate of the CA.    |  
```
```suggestion
| `-out node.csr`     | The public key certificate of the CA.    |  
```
@patrickangeles the code above uses `-out node.csr`. Should I change this to match?

dcoded replied:
Yes, this would be `-out node.csr`.

micheleRP commented:
@patrickangeles, @dcoded: I renamed this as Patrick suggested, I moved introductory info to the top of the page, and I removed comments from the examples so they copy/paste better. Examples now includes <editable-variables> in red, so you can replace them with actual values before you copy it. (If these don't appear red/editable, the page somehow refreshed, and just reload it by toggling to another page and back to see them editable again.)

I added a couple questions. The remaining feedback is good and I don't want to lose it... but I also don't want to hold up the edits on this page from publishing. Unless either of you have can provide more content for the items below, I'd like to merge this PR and put the below items into a new doc issue to follow up separately. Please let me know what you think!

- This covers client-to-broker communications via the Kafka API... I haven't seen how you setup TLS for broker-to-broker RPC communications. Is that covered in another section? (michele: No. I think the only bit we have is here: https://docs.redpanda.com/docs/reference/node-properties/#rpc_server_tls)
- Some sections could use further context. For example: why is this step optional? Optional: Create a local CA for self-signed certificates (michele: I don't know. Can either of you explain that?)
- General comment: overall it reads somewhere between a HOWTO and a reference doc... it might be clearer to break these out into separate pages? Or structure this page somehow so that we go through what needs to be configured, but also show some of the example steps but be clear that they're examples. (michele: The authentication and authorization pages also should be broken into separate conceptual and task pages.)
- Related is that the examples seems to assume a host based installation... you can't just edit redpanda.yaml for a container based install. So the examples won't apply to any user trying this on a docker based setup (which is a likely scenario if you're following a HOWTO and trying this on your local machine). (michele: This is beyond me, as I've only used redpanda with docker. I'd need an SME to help with this.)

micheleRP commented:
To wrap up this original issue, I'm moving remaining open comments into new doc issue: https://github.com/redpanda-data/documentation/issues/1425

micheleRP commented:
merging this, with confirmation from Denis

JakeSCahill commented on tests/setup-tests/.doc-detective.json:
```diff
@@ -63,15 +65,18 @@
+  "telemetry": {
+    "send": true,
+    "userId": "Redpanda Data"
+  }
```
```suggestion
```
